From: Ben Hutchings <ben@decadent.org.uk>
Subject: Revert "wrong type for 'magic' argument in simple_fill_super()"

This is an ABI change and is only required for IMA which we don't enable.

reverted:
--- b/fs/libfs.c
+++ a/fs/libfs.c
@@ -415,8 +415,7 @@
  * unique inode values later for this filesystem, then you must take care
  * to pass it an appropriate max_reserved value to avoid collisions.
  */
+int simple_fill_super(struct super_block *s, int magic, struct tree_descr *files)
-int simple_fill_super(struct super_block *s, unsigned long magic,
-		      struct tree_descr *files)
 {
 	struct inode *inode;
 	struct dentry *root;
reverted:
--- b/include/linux/fs.h
+++ a/include/linux/fs.h
@@ -2376,7 +2376,7 @@
 extern const struct inode_operations simple_dir_inode_operations;
 struct tree_descr { char *name; const struct file_operations *ops; int mode; };
 struct dentry *d_alloc_name(struct dentry *, const char *);
+extern int simple_fill_super(struct super_block *, int, struct tree_descr *);
-extern int simple_fill_super(struct super_block *, unsigned long, struct tree_descr *);
 extern int simple_pin_fs(struct file_system_type *, struct vfsmount **mount, int *count);
 extern void simple_release_fs(struct vfsmount **mount, int *count);
 
